tests/utils.go

```diff
@@ -429,6 +445,122 @@ func CreateSecret(name string, data map[string]string) {
 	}
 }
 
+func GetListOfManifests(pathToManifestsDir string) []string {
+	var manifests []string
+	files, err := ioutil.ReadDir(pathToManifestsDir)
+	if err != nil {
+		panic(err)
+	}
+	matchFileName := func(pattern, filename string) bool {
+		match, err := filepath.Match(pattern, filename)
+		if err != nil {
+			panic(err)
+		}
+		return match
+	}
+	for _, file := range files {
+		if matchFileName("*-for-ocp.yaml", file.Name()) {
+			if CurrentClusterType.value == OpenShiftClusterType {
+				manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+			}
+		} else if matchFileName("*-for-k8s.yaml", file.Name()) {
+			if CurrentClusterType.value == KubernetesClusterType {
+				manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+			}
+		} else if matchFileName("*.yaml", file.Name()) {
+			manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+		}
+	}
+	return manifests
+}
+
+func ReadManifestYamlFile(pathToManifest string) []unstructured.Unstructured {
+	// Read manifest
+	content, err := ioutil.ReadFile(pathToManifest)
+	if err != nil {
+		panic(err)
+	}
+	// Split manifest by `---` to get single objects
+	fileAsString := string(content[:])
+	sepYamlfiles := strings.Split(fileAsString, "---")
+	// Parse each object
+	objects := make([]unstructured.Unstructured, 0, len(sepYamlfiles))
+	for _, f := range sepYamlfiles {
+		if f == "\n" || f == "" {
+			// ignore empty cases
+			continue
+		}
+
+		var obj map[string]interface{}
+		err := yaml.Unmarshal([]byte(f), &obj)
+		if err != nil {
+			fmt.Printf(fmt.Sprintf("ERROR: %s\n", err))
+		}
+		unstruct := unstructured.Unstructured{}
+		unstruct.SetUnstructuredContent(obj)
+		// FIXME: remove these debugs
+		fmt.Printf(fmt.Sprintf("  Kind(%s)\n", unstruct.GetKind()))
+		fmt.Printf(fmt.Sprintf("    %s\n", unstruct))
+
+		objects = append(objects, unstruct)
+	}
+
+	return objects
+}
+
+func ApplyRawManifest(object unstructured.Unstructured) error {
+	virtCli, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+	uri := "/api"
+	if object.GetAPIVersion() != "v1" {
+		uri += "s"
+	}
+	uri += "/" + object.GetAPIVersion()
+	if object.GetNamespace() != "" {
+		uri += "/namespaces/" + object.GetNamespace()
+	}
+	uri += "/" + strings.ToLower(object.GetKind()) + "s"
+	// FIXME: remove these debugs
+	fmt.Printf(fmt.Sprintf("GroupVersionKind: %s\n", object.GroupVersionKind()))
+	fmt.Printf(fmt.Sprintf("GetResourceVersion: %s\n", object.GetResourceVersion()))
+	fmt.Printf(fmt.Sprintf("GetObjectKind: %s\n", object.GetObjectKind()))
+	fmt.Printf(fmt.Sprintf("API URI: %s\n", uri))
+
+	// ### Using pure Unstructured Object
```

Comment:
@rmohr  I have an issue to post body as an Unstructured via API.
Here bellow are three ways which I have tried.

Regarding to json, problem is that it expect `map[string]interface{}` but YAML module loads everything as `map[insterface{}]interface{}`, I could cast this variable but I will have to go even into nested maps and cast them too . Do you think this is correct way ?

Reply:
@lukas-bednar you can simply call `obj.MarshalJSON()`. That returns `[]byte, error`. Just use the bytes as body.

Reply:
oh wait.  You had that already. Let me check.

Reply:
Ha, I see it. Do `DoRaw()` intead of `Do()`.

Reply:
Thanks!